src/test/javascript/spec/component/programming-exercise/programming-exercise-lifecycle.component.spec.ts

```diff
@@ -243,4 +245,43 @@ describe('ProgrammingExerciseLifecycleComponent', () => {
         expect(alertSpy).toHaveBeenCalledTimes(nthCall + 1);
         expect(alertSpy).toHaveBeenNthCalledWith(++nthCall, 'artemisApp.programmingExercise.timeline.alertNewExampleSolutionPublicationDateAsDueDate');
     });
+
+    it.each([true, false])('should enable checkbox for complaints on automatic assessments for automatically assessed exercises', (examMode: boolean) => {
+        comp.exercise = exercise;
+        comp.exercise.assessmentType = AssessmentType.AUTOMATIC;
+        comp.isExamMode = examMode;
+        if (!examMode) {
+            exercise.dueDate = undefined;
+            exercise.course = new Course();
+            exercise.course.complaintsEnabled = true;
+        }
+        fixture.detectChanges();
+        const checkbox: HTMLInputElement = fixture.debugElement.nativeElement.querySelector('#allowComplaintsForAutomaticAssessment');
+        expectElementToBeEnabled(checkbox);
+    });
+
+    it.each([true, false])('should disable checkbox for complaints on automatic assessments for exercises without automatic assessment', (examMode: boolean) => {
+        comp.exercise = exercise;
+        comp.exercise.assessmentType = AssessmentType.SEMI_AUTOMATIC;
+        comp.isExamMode = examMode;
+        if (!examMode) {
+            exercise.course = new Course();
+            exercise.course.complaintsEnabled = true;
+        }
+        fixture.detectChanges();
+        const checkbox: HTMLInputElement = fixture.debugElement.nativeElement.querySelector('#allowComplaintsForAutomaticAssessment');
+        expectElementToBeDisabled(checkbox);
+    });
+
+    it('should disable checkbox for complaints on automatic assessments for automatically assessed course exercises without due date', () => {
```

Comment:
Where is the difference in the set up to the first new test with examMode = false?

Reply:
Thanks for noticing this flaw in my test logic. Should be fixed now.